### DISCUSSION TITLE
Sprite Pro Install and Firmware Tuning

### DISCUSSION THREAD

**Discussion by FlyBoy33** [mriscoc/Ender3V2S1#909] [Q&A]
Hello again! PLEASE bare with me!

I;ve installed the firmware. Now I want to install a Sprite Pro Extruder on my E3V2 Neo. What, and How do I need to make the adjustments for this extruder, and adjustments for the relocation for the CR Touch??
You guys/gals are so much help and I really want to get a better understanding of how to make, and what all the adjustments do, once the firmware has been installed. Is there any comprehensive guide to what each adjustment will do? Thank you! Forever grateful!

**Comment by Crazy-Charles** [comment]
The nozzle to probe offsets for the sprite and cr-touch are -31.8 for X and -40-5 for Y, change them using the "Probe Settings" menu and dont forget to "save settings" afterwards.

https://github.com/mriscoc/Ender3V2S1/wiki is a good place to start

**Comment by houstnwehavuhoh** [comment]
You will need to adjust offsets for Sprite and CR Touch. As a starting point, here are my settings. You'll likely need to fine tune for your machine. 

An easy way to determine accuracy is to apply painters tape over your bed, measure and draw an X from each corner, determine the probe and nozzle home at the center of the X, and then after saving mesh insert dimensions, use the tramming wizard to make sure the 4 corners are accurately probed by if they fall on each line of the X. It is worth making sure that X is drawn as accurate as possible, though not necessary.

Printer bed sizing:

- X Bed Size: 220
- Y Bed Size: 220
- X Min Position: 0
- Y Min Position: -6
- X Max Position: 233
- Y Max Position: 225
- Z Max Position: 240

Probe Settings:

- Probe X Offset: -33.0
- Probe Y Offset: -42.0

Mesh Insert:

- Mesh X Min: 37.0
- Mesh X Max: 183
- Mesh Y Min: 37
- Mesh Y Max: 183

Note that the mesh insert is actually auto calculated. Any dimensional changes would of course change the mesh, so make sure you update it after - select "Maximize Mesh", then "Center Mesh". Don't forget to store settings. Also note that the Sprite reduces the bed size due to being dimensionally larger on the x gantry, and your mesh area will be smaller, meaning your effective bed size is even smaller. Not a big deal, just worth noting.

There is a guide on this GitHub of how to dial this in, which will give you a better understanding for fine tuning. Don't forget the basics - esteps, MPC and PID tune, z offset, creating mesh, updating start gcode, etc. Good luck

**Reply by darraghh1** [reply]
For anyone who noted that their head was impacting the bed, this answer 100% solved my problem.

Ender 3 V2, 427  Motherboard, custom firmware (CreateConfigs.Generate('MyConfiguration',['Ender3V2','422','BLT','UBL','T13','MPC']), Sprite Pro, CR Touch.

**Reply by darraghh1** [reply]
@houstnwehavuhoh Thank you for pointing me in the right direction!
